app/views/admin/calendars/_import.html.erb

```diff
@@ -3,7 +3,11 @@
 
 <%= render 'status' %>
 
-<% if @calendar.calendar_state.idle? %>
+<% if @calendar.calendar_state.idle? || @calendar.calendar_state.bad_source? %>
```

Comment:
This is the only bit I'm unsure about - we don't do this for errored calendars but we do allow them to be queued for import - I just wasn't sure what the associated risks were with letting someone re-run this. Especially if they've changed the URL, for example, it feels like we should let them rerun?

Reply:
If they change the URL the calendar should be forced into the queue again (?) (or should be)

Reply:
But in this bug, the URL is fine & their calendar is listed as a bad_source despite being fine and they can't do anything about it

Reply:
This seems okay. As a non-blocking suggestion it may be an idea to move this logic out of the view and into the model as a `can_be_requeued?` method so it can be unit-tested.

Reply:
I've moved the logic into the model and refactored one other line to make it clearer what is being checked (whether a calendar is busy or not). I'm not going to write a unit test for this at this point because it would be super closely coupled to this specific method, but it's clearer from its name what it should be doing which is very helpful!